wireguard-go-rust-wrapper/wireguard-go/build.ps1

```diff
@@ -6,10 +6,21 @@ $ErrorActionPreference = "Stop"
 # cannot be linked with the Rust output, since Rust uses MSVC and its libraries.
 #
 # So let CLang from Visual Studio 2019 and ar.exe from TDM-GCC be our knights in shining armor.
-$env:CC="clang.exe";
+$arch = $Args[0];
+
+$env:CGO_ENABLED = "1";
+$env:GOARCH = $arch;
+
+if ($arch -eq "amd64") {
+    $env:CC="clang.exe --target=x86_64-pc-windows-msvc";
+} elseif ($arch -eq "arm64") {
+    $env:CC="clang.exe --target=arm64-pc-windows-msvc";
+} else {
+    throw "Unsupported architecture: $arch";
+}
 
 # WORKAROUND: Rust on Windows will look for LIBs without "lib" prefix, so we're naming ours "wireguard-go.lib"
-& go build -buildvcs=false -ldflags=-w -v -buildmode c-archive -o "$env:OUT_DIR\wireguard-go.lib" wireguard-go
+go build -buildvcs=false -ldflags=-w -v -buildmode c-archive -o "$env:OUT_DIR\wireguard-go.lib" wireguard-go
```

Comment:
Idk much about powershell scripts, but why you remove this `&`?

Reply:
Honestly, I do not remember now, cause it was ~4 months ago :smile:  I'll try without now and see if it works.

Reply:
It seems that it is good to run with `&` in front, so will leave it here.